src/GameLogic/DataStructures/EntityMovement.ts

```diff
@@ -0,0 +1,24 @@
+import {Entity} from '../Entity';
+
+export class EntityMovement {
+	public entity: Entity;
+
+	public oldX: number;
+
+	public oldY: number;
+
+	public newX: number;
+
+	public newY: number;
+
+	public others: EntityMovement[];
+
+	constructor(entity: Entity, oldX: number, oldY: number, newX: number = oldX, newY: number = oldY) {
```

Comment:
`newX: number = oldX, newY: number = oldY` What is this magic, you can do that and it works? :O

Reply:
Javascript: Everything works